dotCMS/src/main/java/com/dotcms/content/elasticsearch/util/ESMappingUtilHelper.java

```diff
@@ -298,6 +301,61 @@ private void addMappingForFieldIfNeeded(final String indexName,
         }
     }
 
+    /**
+     * Given a {@link Field}, obtains the ES mapping according to the {@link Field} type
+     * @param field
+     * @param fieldVariableName
+     * @return
+     */
+    private String getMappingForField(final Field field, final String fieldVariableName) {
+        String mappingForField = null;
+        if (field instanceof DateField || field instanceof DateTimeField
+                || field instanceof TimeField) {
+            mappingForField = "{\n\"type\":\"date\",\n";
+            mappingForField += "\"format\": \"yyyy-MM-dd't'HH:mm:ss||MMM d, yyyy h:mm:ss a||yyyy-MM-dd HH:mm:ss||yyyy-MM-dd||epoch_millis\"\n}";
```

Comment:
Codacy found an issue: [Prefer StringBuilder (non-synchronized) or StringBuffer (synchronized) over += for concatenating strings](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5866177)